### PR TITLE
Added page about modules

### PR DESCRIPTION
**Pull request by igorkorsukov**

### CODE REVIEW

File: Architecture/Modularity.md [hunk 1]
```diff
@@ -0,0 +1,134 @@
...
+    void onInit() override;
+};
+}
+}
+```
+
+Possible methods see in `imodulesetup.h`, also see examples of setups of other modules.
+
+The module setup must be added to the list of modules in `main/modulessetup.cpp`  
+And module must be added to list of linking in `main/CMakeLists.txt`
```
Comment (Eism) [inline Architecture/Modularity.md:102]: We can add a cmake flag to enable or disable the module. If the module will be truly standalone.

Comment (igorkorsukov) [reply Architecture/Modularity.md:102]: Yes, this is preferable, I hope in the future most of the modules will be disabled, primarily to optimize development.

File: Architecture/Modularity.md [hunk 2]
```diff
@@ -0,0 +1,134 @@
...
+terms of the problem domain, rather than in terms of low-level implementation
+details, and working at the highest level of abstraction reduce the load on your brain.  
+The bottom line is that programmers who compensate for inherent human limitations write code that’s easier for themselves and others to understand and that has
+fewer errors.
+
+(c) Steven C. McConnell "Code Complete" Chapter 5
+
+## Modules
+
+We divide the entire application into a module.
```
Comment (vpereverzev) [inline Architecture/Modularity.md:29]: modules

File: Architecture/Modularity.md [hunk 3]
```diff
@@ -0,0 +1,134 @@
...
+A module is a functional unit that, on the one hand, contains independent value, on the other hand, an application can have value without this module.
```
Comment (vpereverzev) [inline Architecture/Modularity.md:30]: an application **can have value** without this module

I'd suggest to replace it by "can exists"/"can work"/"may be beneficial"

File: Architecture/Modularity.md [hunk 4]
```diff
@@ -0,0 +1,134 @@
...
+}
+}
+```
+
+Possible methods see in `imodulesetup.h`, also see examples of setups of other modules.
+
+The module setup must be added to the list of modules in `main/modulessetup.cpp`  
+And module must be added to list of linking in `main/CMakeLists.txt`
+
+### Public interfaces
```
Comment (vpereverzev) [inline Architecture/Modularity.md:104]: Well, I think we also need to explain the benefits of this approach

Comment (igorkorsukov) [reply Architecture/Modularity.md:104]: Explain the advantages of using interfaces or the fact that they are located in the root directory of the module?

File: Architecture/Modularity.md (hunk 2 above)
Comment (igorkorsukov) [reply Architecture/Modularity.md]: fixed

File: Architecture/Modularity.md (hunk 3 above)
Comment (igorkorsukov) [reply Architecture/Modularity.md]: replaced on "can work"

File: Architecture/Modularity.md (hunk 4 above)
Comment (vpereverzev) [reply Architecture/Modularity.md:104]: explain the second one

Comment (igorkorsukov) [reply Architecture/Modularity.md:104]: done

**Pull request merged**